apps/rule-manager/app/db/DbRuleDraft.scala

```diff
@@ -276,6 +276,42 @@ object DbRuleDraft extends SQLSyntaxSupport[DbRuleDraft] {
       case Left(result) => Left(result)
     }
   }
+  def batchUpdateFromFormRule(ids: List[Int], category: String, tags: String, user: String)(implicit
+      session: DBSession = autoSession
+  ): Try[List[DbRuleDraft]] = {
+    Try {
+      val updatedRows = withSQL {
+        update(DbRuleDraft)
+          .set(
+            column.category -> category,
+            column.tags -> tags,
+            column.updatedBy -> user,
+            column.revisionId -> sqls"${column.revisionId} + 1"
+          )
+          .where
+          .in(column.id, ids)
+      }.update().apply()
+
+      if (updatedRows > 0) {
+        ids.flatMap { id =>
+          find(id) match {
```

Comment:
The update clause looks perfect, but we're running into the same problem with this select clause – it'll run once for every rule, which on an update of thousands will be very slow.

Instead, could we try to select many rules by id? We could write a new version of `findAll`, perhaps, that accepted a list of ids, and used a `where.in(...)` clause to select the relevant rules. This'd be nice and quick.

This should mean that we don't have to worry about the rehydration of categories/tags on L301/2 (although I'm not quite sure w

Reply:
That's a good point, thanks for raising it 👍